### ISSUE TITLE
Opting in/out of certain Twig functions & filters

### ISSUE THREAD

**Issue by joelambert** [timber/timber#2192]
## Is your feature request related to a problem? Please describe.
Much of the work we do requires our Twig templates to be not too closely tied to Timber. Specifically, we share the templates between a Design System and WordPress.

As such, we don't as a rule make use of Timber specific filters and functions inside Twig, instead favouring more Timber work in our Controllers. This presents:

1. a potential challenge with developers making use of these less portable filters
2. an opportunity to reduce how much needs to be loaded on each request

## Describe the solution you’d like
I realise this is possibly a niche requirement but is there any appetite for a more configurable way to load the filters and functions?

Rough idea would be to modify `Twig::add_timber_functions()` and `Twig::add_timber_filters()` function to be more extendable. If these functions called private functions that return and associative array of keys and callables this could then potentially be filtered before adding to Twig.

Pseudo code example:

```
public function add_timber_functions( $twig ) {
    $functions = $this->get_timber_functions();

    $functions = apply_filters('timber/twig/functions/add`, $functions);

    foreach ($functions as $key => $callable) {
        $twig->addFunction(new Twig_Function($key, $callable); 
    }
}

protected function get_timber_functions() {
    $functions = [];

    $functions['bloginfo'] = 'bloginfo';
    $functions['foo'] = function () {};

    return $functions;
}
``` 

Is there any appetite for this sort of a change? If so I can put together a PR for it.

**Comment by jarednova** [comment]
@joelambert I'm fine with a PR for this philosophically. But before going too far down the rabbit hole, I'm wondering if `remove_action`:

https://developer.wordpress.org/reference/functions/remove_action/

... could be used to target these particular functions within `Timber\Twig` so they don't interfere with your app

**Comment by nlemoine** [comment]
I also think that filtering Timber filters & functions would be interesting. The issue with `remove_action` is that it's an all or nothing solution: you can only remove **all** functions/filters. 

Furthermore, Timber [makes it hard to apply the `remove_action`](https://github.com/timber/timber/blob/master/lib/Twig.php#L20) because methods are not `static` and the `Timber\Twig` object isn't publicly accessible.

I would also be in favor of filter like the one proposed by @joelambert.

Let me know if that fits for you, I can work on a PR.

**Comment by gchtr** [comment]
Resolved in #2408. Big thanks to @nlemoine for leading the way! 🙌
